hack/e2e_compare.sh

```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+set -e
+
+# Compare 2 files generated by e2e_collect. This ensures elements are sorted by version
+
+type -f jq > /dev/null || {
+  echo Please install jq
+  exit 1
+}
+
+# jq would output the whole graph with elements in "nodes" sorted by "version"
+# to ensure the json output is identical for the equal graphs
+diff <(jq -S '.nodes |= sort_by(.version)' ${1}) <(jq -S '.nodes |= sort_by(.version)' ${2}) || exit 0
```

Comment:
I suggest not to mask the exit state here and leave this decision to the caller. If the caller wants to treat this as an error, it'd have to rely on the callee's stdout instead to see if there was a difference.

Reply:
Okay, makes sense.

Seems default diff exit codes would work here too - `Exit status is 0 if inputs are the same, 1 if different, 2 if trouble`